server/src/main/java/org/elasticsearch/search/aggregations/MultiBucketConsumerService.java

```diff
@@ -52,27 +53,35 @@ private void setMaxBucket(int maxBucket) {
 
     public static class TooManyBucketsException extends AggregationExecutionException {
         private final int maxBuckets;
+        private final int bucketsCount;
 
-        public TooManyBucketsException(String message, int maxBuckets) {
+        public TooManyBucketsException(String message, int maxBuckets, int bucketsCount) {
             super(message);
             this.maxBuckets = maxBuckets;
+            this.bucketsCount = bucketsCount;
         }
 
         public TooManyBucketsException(StreamInput in) throws IOException {
             super(in);
             maxBuckets = in.readInt();
+            bucketsCount = in.getTransportVersion().onOrAfter(TransportVersions.LOG_TOO_MANY_BUCKETS_ERROR) ? in.readVInt() : -1;
         }
 
         @Override
         protected void writeTo(StreamOutput out, Writer<Throwable> nestedExceptionsWriter) throws IOException {
             super.writeTo(out, nestedExceptionsWriter);
             out.writeInt(maxBuckets);
+            out.writeVInt(out.getTransportVersion().onOrAfter(TransportVersions.LOG_TOO_MANY_BUCKETS_ERROR) ? bucketsCount : -1);
```

Comment:
I think this should be: 

```
if (out.getTransportVersion().onOrAfter(TransportVersions.LOG_TOO_MANY_BUCKETS_ERROR)) {
    out.writeVInt(bucketsCount);
}
```

Otherwise `writeVint()` invocation fails?